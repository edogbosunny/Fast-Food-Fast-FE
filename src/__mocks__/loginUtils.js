/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
