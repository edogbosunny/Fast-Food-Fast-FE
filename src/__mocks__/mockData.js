const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
