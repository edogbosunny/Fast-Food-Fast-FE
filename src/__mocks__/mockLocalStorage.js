let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
