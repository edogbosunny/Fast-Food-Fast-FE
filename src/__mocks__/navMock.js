/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
