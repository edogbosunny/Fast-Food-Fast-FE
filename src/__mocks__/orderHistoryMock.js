/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
