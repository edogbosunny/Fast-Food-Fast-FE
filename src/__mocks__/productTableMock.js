/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
