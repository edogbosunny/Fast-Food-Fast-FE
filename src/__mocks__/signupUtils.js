/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  loader: {
    loading: false,
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  registerUser: jest.fn(),
};

export const mockedSignupState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },

  loader: {
    loading: false,
  },
};
