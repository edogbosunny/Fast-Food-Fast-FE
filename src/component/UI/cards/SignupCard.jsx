/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import '../../../styles/style.css';
import TextFieldGroup from '../textField/TextFieldGroup.jsx';
import { registerUser } from '../../../actions/authActions';
import Spinner from '../../../utils/Spinner';

class SignupCard extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      firstname: '',
      lastname: '',
      confirmpassword: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    if (this.props.auth.isAuthenticated) {
      // console.log(this.props);
      // eslint-disable-next-line react/destructuring-assignment
      this.props.history.push('/');
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const {
      firstname, lastname, email, password, confirmpassword,
    } = this.state;
    const newUser = {
      firstname,
      lastname,
      email,
      password,
      confirmpassword,
    };
    const { registerUser, history } = this.props;
    registerUser(newUser, history);
    // console.log('newuser==>', newUser);
  }

  render() {
    const {
      firstname, lastname, confirmpassword, email, password,
    } = this.state;
    const { errors, loader } = this.props;
    const { loading } = loader;
    if (loading) {
      return (<Spinner />);
    }
    return (
      <div className='Aligner '>
        <div className='loginCard '>
          <div className='aligncnt '>
            <h2 className='mt'>Create Account </h2>
            {/* <!-- <h4 style="margin-top:-25px; color:#f2f2f2 ">
        Get your food in minutes</h4> --> */}
            <form noValidate onSubmit={this.onSubmit}>
              <div className='divwidth '>
                <div>
                  <TextFieldGroup
                    placeholder='First Name'
                    name='firstname'
                    value={firstname}
                    type='firstName'
                    error={errors.firstname || ''}
                    onChange={this.onChange}
                    minLength='5'
                  />

                  <TextFieldGroup
                    placeholder='Last Name'
                    name='lastname'
                    type='lastname'
                    error={errors.lastname || ''}
                    value={lastname}
                    onChange={this.onChange}
                  />

                  <TextFieldGroup
                    placeholder='email'
                    name='email'
                    type='email'
                    error={errors.email || ''}
                    value={email}
                    onChange={this.onChange}
                  />

                  <TextFieldGroup
                    placeholder='Password'
                    name='password'
                    type='password'
                    error={errors.password || ''}
                    value={password}
                    onChange={this.onChange}
                  />

                  <TextFieldGroup
                    placeholder='Confirm Password'
                    name='confirmpassword'
                    type='confirmpassword'
                    error={errors.confirmpassword || ''}
                    value={confirmpassword}
                    onChange={this.onChange}
                  />

                  <small>{errors.message}</small>
                </div>
                <div>
                  <input
                    type='submit'
                    className=' submitbtn btn btn-info btn-block mt-4'
                  />
                </div>
              </div>

              <div className='bm' />
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors,
  loader: state.loader,
});

export default withRouter(connect(mapStateToProps, { registerUser })(SignupCard));
