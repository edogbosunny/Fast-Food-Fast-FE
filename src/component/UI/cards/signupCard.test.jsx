import React from 'react';
import { mount } from 'enzyme';
import { SignupCard, mapStateToProps } from './SignupCard';
import { props, mockedSignupState } from '../../../__mocks__/signupUtils';

describe('Signup card component', () => {
  const wrapper = mount(<SignupCard {...props} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for onchange', () => {
    const event = {
      target: { name: 'email', value: 'test@test.com' },
      preventDefault: () => { },
    };

    wrapper.instance().onChange(event);
    expect(wrapper.state().email).toEqual('test@test.com');
    expect(wrapper.instance().state.email).toEqual(event.target.value);
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(mockedSignupState);
    expect(state).toBeTruthy();
  });

  it('should test the onSubmit function', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    const loginForm = wrapper.find('form');
    loginForm.simulate('submit', fakeEvent);
    const registerUser = jest.fn(mockedSignupState);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().onSubmit);
    });
    promise.then(() => expect(registerUser).toHaveBeenCalledTimes(1));
  });
});
