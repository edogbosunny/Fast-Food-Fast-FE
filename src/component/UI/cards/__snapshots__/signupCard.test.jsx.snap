// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup card component should match snapshot 1`] = `
<SignupCard
  auth={
    Object {
      "isAuthenticated": true,
      "loading": null,
      "user": Object {
        "exp": 1549525677,
        "iat": 1549439277,
        "id": 1,
      },
    }
  }
  errors={
    Object {
      "email": "",
      "isAuthenticated": false,
      "loading": null,
      "message": "",
      "password": "",
    }
  }
  history={
    Object {
      "action": "PUSH",
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    }
  }
  loader={
    Object {
      "loading": false,
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/login",
      "search": "",
    }
  }
  registerUser={[MockFunction]}
  user={Object {}}
>
  <div
    className="Aligner "
  >
    <div
      className="loginCard "
    >
      <div
        className="aligncnt "
      >
        <h2
          className="mt"
        >
          Create Account 
        </h2>
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="divwidth "
          >
            <div>
              <TextFieldGroup
                error=""
                minLength="5"
                name="firstname"
                onChange={[Function]}
                placeholder="First Name"
                type="firstName"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    minLength="5"
                    name="firstname"
                    onChange={[Function]}
                    placeholder="First Name"
                    required={true}
                    type="firstName"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <TextFieldGroup
                error=""
                name="lastname"
                onChange={[Function]}
                placeholder="Last Name"
                type="lastname"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    name="lastname"
                    onChange={[Function]}
                    placeholder="Last Name"
                    required={true}
                    type="lastname"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <TextFieldGroup
                error=""
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <TextFieldGroup
                error=""
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <TextFieldGroup
                error=""
                name="confirmpassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="confirmpassword"
                value=""
              >
                <div
                  className="txtwidth form-style-1"
                >
                  <input
                    className="form-control"
                    name="confirmpassword"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    required={true}
                    type="confirmpassword"
                    value=""
                  />
                </div>
              </TextFieldGroup>
              <small />
            </div>
            <div>
              <input
                className=" submitbtn btn btn-info btn-block mt-4"
                type="submit"
              />
            </div>
          </div>
          <div
            className="bm"
          />
        </form>
      </div>
    </div>
  </div>
</SignupCard>
`;
